edi_835_parser/loops/transaction.py

```diff
@@ -170,18 +183,28 @@ def build(
 					transaction.provider_summary = provider_summary
 					segment = None
 
+				elif identifier == HeaderNumber.identification:
+					header_number = HeaderNumber(segment)
+					transaction.header_number = header_number
+					segment = None
+
+				elif identifier == ReferenceSegment.identification:
+					reference = ReferenceSegment(segment)
+					transaction.references.append(reference)
+					segment = None
+
 				elif identifier == DateSegment.identification:
-						date = DateSegment(segment)
-						transaction.date = date
-						segment = None
+					date = DateSegment(segment)
+					transaction.date = date
+					segment = None
 
 				elif identifier in cls.terminating_identifiers:
 					return transaction, segments, segment
 
 				else:
-					segment = None
-					message = f'Identifier: {identifier} not handled in transaction loop.'
+					message = f'Identifier: {identifier} not handled in transaction loop. {segment}'
```

Comment:
I found seeing the segment right away **VERY** useful to go check specs.

Reply:
Freaking neeeeeeeeed a better pattern here. But yes, this is gud 👍